### ISSUE TITLE
[4.x] Keeping Bard's save_html option active is glitchy

### ISSUE THREAD

**Issue by jacksleight** [statamic/cms#8042]
### Bug description

The `Bard::shouldSaveHtml()` method checks if the `sets` option is empty before allowing `save_html`, but now that the sets are grouped there's almost always an empty group so the check is never false and `save_html` doesn't get enabled.

You can delete the default group, but when setting up a new field it's not obvious you'd need to do that, and it comes back whenever you reopen the field editor.

### How to reproduce

Create a Bard field with `save_html` enabled.

### Environment

```yaml
Statamic
Addons: 0
Antlers: runtime
Stache Watcher: Enabled
Static Caching: Disabled
Version: 4.0.0-beta.2 Solo
```

### Installation

Fresh statamic/statamic site via CLI

### Antlers Parser

None

**Comment by jasonvarga** [comment]
>  the sets are grouped there's almost always an empty group

This is the bug. It shouldn't default to an empty group. It should default to completely empty.

Looking into this. 👍

**Comment by jasonvarga** [comment]
Fixed by #8049
